### PR TITLE
Handle and title updated together

### PR DESCRIPTION
**Pull request by Akarshit**
This PR
1. Makes sure when the product is added, the handle created for it is new and unique
2. When a user manually updates the handle, `copy-<'' or some-number>` would be added to it (if required) to keep it unique.

Fix for #2831 #2560 

To test ->
1. run `reaction`
2. Create a product named "Test Product" and publish it.
3. Open the Product settings, the permalink would be `test-product`
4. Create another product and set its `Title` to "Test Product", the permalink this time should be `test-product-copy`
5. Publish the product
6. Create another product with Title "Test Product". It's permalink would be `test-product-copy-1`. Edit it manually to `test-product`. The permalink should change to `test-product-copy-1` automatically.

Suggestion ->
Maybe we can change the `copy` to `item`. So names would be like `test-product`, `test-product-item`, `test-product-item-1`, ...

### CODE REVIEW

**Comment by aaronjudd** [comment]
@mikemurray can you review this for a potential cherry-pick to `marketplace`?

**Comment by Akarshit** [comment]
@aaronjudd updated for `marketplace`. Also is `marketplace` the main development branch now?

File: server/methods/catalog.js [hunk 1]
```diff
@@ -876,10 +876,23 @@ Meteor.methods({
       const booleanValue = (value === "true" || value === true);
       update = EJSON.parse("{\"" + field + "\":" + booleanValue + "}");
     } else {
-      const stringValue = EJSON.stringify(value);
-      update = EJSON.parse("{\"" + field + "\":" + stringValue + "}");
+      if (field === 'handle') {
```
Comment (mikemurray) [inline server/methods/catalog.js]: Our linting rules require double quotes for all strings.

**Comment by Akarshit** [comment]
@mikemurray Can you please elaborate on how the case was failing.

**Comment by mikemurray** [comment]
1. Create a product
2. Set the title to "Test Product"
3. Click Publish
4. Observe error "label is required", don't fix it
5. Look at the product, and see the slug say "test-product"
6. Create another product
7. Set title to "Test Product"
8. Click publish
7. Observe error "label is required", don't fix it
9. Look at the product, and see the slug say "test-product"
10. Fix the "label is required" error on the variants of both products and publish
11. Now 2 products have the same slug

**Comment by spencern** [comment]
@mikemurray we need to resolve this issue. Either updating / finishing this issue, or closing and fixing in another PR

**Comment by mikemurray** [comment]
@spencern @Akarshit This ticket fixes part of the handle issue and is good to go as is. 

I believe the issue I pointed out has something to do with revision controled documents and is better tracked with another issue. We can continue the discussion in ticket https://github.com/reactioncommerce/reaction/issues/3007 .

**Comment by spencern** [comment]
Great, thanks @mikemurray

**Pull request merged**